argus/backend/service/results_service.py

```diff
@@ -250,21 +297,64 @@ def calculate_graph_ticks(graphs: List[Dict]) -> dict[str, str]:
     return {"min": min_x[:10], "max": max_x[:10]}
 
 
+def _identify_most_changed_package(packages_list: list[PackageVersion]) -> str:
+    version_date_changes: dict[str, set[tuple[str, str]]] = defaultdict(set)
+
+    for package_version in packages_list:
+        version_date_changes[package_version.name].add((package_version.version, package_version.date))
+
+    return max(version_date_changes, key=lambda k: len(version_date_changes[k]))
+
+
+def _split_results_by_release(packages: dict[str, list[PackageVersion]], main_package: str) -> ReleasesMap:
+    releases_map = defaultdict(list)
+    for run_id, package_versions in packages.items():
+        for package in package_versions:
+            if package.name == main_package:
+                if "dev" in package.version:
+                    major_version = 'dev'
+                else:
+                    major_version = '.'.join(package.version.split('.')[:2])
+                releases_map[major_version].append(run_id)
+    return releases_map
+
+
+def create_chartjs(table: ArgusGenericResultMetadata, data: list[ArgusGenericResultData], best_results: dict[str, List[BestResult]],
```

Comment:
Regarding charts in general - could you try enabling cubic interpolation for the line, `rect` style for the points (or just make them tiny dots) and `filled` style for the line itself? (with some opacity thrown in for the color) 

The random data kinda showcases worst case scenario and it looks completely unreadable, especially with a lot of data points. Suggestions above could help with that, I feel.

Reply:
I try to play a bit with it and with grid uniformity too.